src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+
+type Props = {
+  currentTodo: Todo;
+  close: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ currentTodo, close }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getUser(currentTodo.userId)
+      .then(setUser)
+      .catch(() => setUser(null))
```

Comment:
Currently, when fetching user data fails, `setUser(null)` is called. Consider providing more informative feedback to the user, such as displaying an error message, to improve the user experience.